### ISSUE TITLE
Point update safety

### ISSUE THREAD

**Issue by dmadisetti** [jlblancoc/nanoflann#116]
Just from reading over the code, it seems like updating the values within the data-source may potentially break searches. Am I missing something?

The obvious solution to doing "updates", would be to remove and then add a new point, but if a point moves in it's local neighborhood, surely we can work upwards from the leaf to the best branch in a much more efficient manner.

I'd be happy to make a refresh_point function, that will reposition a point in a tree after an update. I will benchmark this against against adding and removing a point method I just mentioned. Let me know if you have any pointers or concerns regarding this

**Comment by yetigit** [comment]
I think it really could break the index, + yeah a explicit update point , or setpoint (indice) is much needed

**Comment by jlblancoc** [comment]
@dmadisetti Yes, you're right: by design, nanoflann data source model explicitly uses references to the source and *assumes* that it's static.

I think your approach might work, so a PR with such a feature would be more than welcome! :-)
